packages/eslint-plugin/src/rules/require-localize-metadata.ts

```diff
@@ -6,29 +6,68 @@ export type Options = [
   {
     readonly requireDescription?: boolean;
     readonly requireMeaning?: boolean;
+    readonly requireCustomId?: boolean;
+    readonly requireDefaultValue?: boolean;
+    readonly boundTextAllowedPattern?: string;
   },
 ];
 
 const DEFAULT_OPTIONS: Options[number] = {
   requireDescription: false,
   requireMeaning: false,
+  requireCustomId: false,
+  requireDefaultValue: false,
 };
 
-const VALID_LOCALIZED_STRING_WITH_DESCRIPTION = new RegExp(
-  /:(.*\|)?([\w\s]+){1}(@@.*)?:.+/,
-);
+export interface I18NMetadata {
+  meaning: boolean;
+  description: boolean;
+  customKey: boolean;
+  defaultValue: boolean;
+}
 
-const VALID_LOCALIZED_STRING_WITH_MEANING = new RegExp(
-  /:([\w\s]+\|)(.*)?(@@.*)?:.+/,
-);
+const validate = (
+  input: string,
+  typeToValidate: string,
+  boundTextAllowedPattern?: string,
+): boolean => {
+  const regex =
+    /^(?::(?:(?<meaning>[^|]+)\|)?(?<description>[^@:]*)?(?:@@(?<customId>[^:]*))?(?::(?<defaultValue>.*))?)$/;
+
+  const match = regex.exec(input.trim());
+
+  let result = false;
+  if (/^(:[^:].[^|:@]*)$/.test(input) === false) {
+    if (
+      typeToValidate === 'defaultValue' &&
+      !input.includes(':') &&
+      !input.includes('|') &&
+      !input.includes('@@')
+    ) {
+      // Input is plain text without special markers
+      result = true;
+    } else if (match && match.groups) {
+      if (typeToValidate === 'customId' && boundTextAllowedPattern) {
+        const boundTextAllowedRegex = new RegExp(boundTextAllowedPattern);
+        return boundTextAllowedRegex.test(match.groups[typeToValidate]);
+      } else {
+        result = !!match.groups[typeToValidate];
+      }
+    }
+  }
+  return result;
+};
 
 const STYLE_GUIDE_LINK = 'https://angular.dev/guide/i18n';
 const STYLE_GUIDE_LINK_COMMON_PREPARE = `${STYLE_GUIDE_LINK}-common-prepare`;
```

Comment:
This link no longer seems to be correct at all (I think the new one is https://angular.dev/guide/i18n/prepare?) so let's please fix it as part of this PR so that all the various errors have reasonable links.

I discovered this when trying to verify the ones you added (which should be here now I believe https://angular.dev/guide/i18n/manage-marked-text)

Reply:
Updated.